bundles/org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/preferences/GenericEditorPreferencePage.java

```diff
@@ -0,0 +1,240 @@
+		button.setData(meta);
+		button.setText(meta.name());
+		button.setToolTipText(meta.description());
+		buttons.put(meta, button);
+		return button;
+	}
+
+	private Control[] createTextField(final PreferenceMetadata<String> meta, Composite composite, int textLimit,
+			int horizontalIndent) {
+		Label labelControl = new Label(composite, SWT.NONE);
+		labelControl.setText(meta.name());
+		GridData gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		gd.horizontalIndent = horizontalIndent;
+		labelControl.setLayoutData(gd);
+
+		final Text textControl = new Text(composite, SWT.BORDER | SWT.SINGLE);
+		gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		gd.widthHint = convertWidthInCharsToPixels(textLimit + 1);
+		textControl.setLayoutData(gd);
+		textControl.setTextLimit(textLimit);
+		textControl.setToolTipText(meta.description());
+
+		textControl.addModifyListener(e -> {
+			updateStatus(validateDelay(textControl.getText()));
+		});
+		textFields.put(meta, textControl);
+		return new Control[] { labelControl, textControl };
+	}
+
+	private static String validateDelay(String value) {
+		if (value.isEmpty()) {
+			return Messages.ContentAssistant_autoActivationDelay_EmptyInput;
+		}
+		try {
+			int integer = parseInteger(value);
+			if (integer < 0)
+				return NLS.bind(Messages.ContentAssistant_autoActivationDelay_InvalidInput, integer);
+		} catch (NumberFormatException e) {
+			return NLS.bind(Messages.ContentAssistant_autoActivationDelay_InvalidInput, value);
+		}
+		return null;
+	}
+
+	private void updateStatus(String errorMessage) {
+		setValid(errorMessage == null);
+		var messageType = errorMessage == null ? IMessageProvider.NONE : IMessageProvider.ERROR;
+		setMessage(errorMessage, messageType);
+		setErrorMessage(errorMessage);
+	}
+
+	private static int parseInteger(Object value) throws NumberFormatException {
+		if (value instanceof Integer) {
+			return ((Integer) value).intValue();
+		}
+		if (value instanceof String) {
+			return Integer.parseInt((String) value);
+		}
+		throw new NumberFormatException(NLS.bind(Messages.ContentAssistant_autoActivationDelay_InvalidInput, value));
+	}
+
+	private void createDependency(final Button lead, String leadKey, final Control[] follower) {
+		var leadState = store.getBoolean(leadKey);
+		for (Control f : follower) {
+			f.setEnabled(leadState);
+		}
+
+		SelectionListener listener = new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				boolean state = lead.getSelection();
+				for (Control f : follower) {
+					f.setEnabled(state);
+				}
+			}
+		};
+		leadFollowerListeners.add(listener);
+		lead.addSelectionListener(listener);
+	}
+
+	private void updateFollower() {
+		Iterator<SelectionListener> iter = leadFollowerListeners.iterator();
+		while (iter.hasNext()) {
+			SelectionListener listener = iter.next();
+			listener.widgetSelected(null);
+		}
```

Comment:
Nit-pick: convert to enhanced for-loop and inline variable:
```suggestion
		for (SelectionListener listener : leadFollowerListeners) {
			listener.widgetSelected(null);
		}
```